scapy/contrib/cansocket_native.py

```diff
@@ -74,6 +79,47 @@ def __init__(self, channel=None, receive_own_messages=False,
         self.ins.bind((self.channel,))
         self.outs = self.ins
 
+    def command(self):
+        d = dict(
+            channel=self.channel,
+            receive_own_messages=self.receive_own_messages,
+            can_filters=self.can_filters,
+            remove_padding=self.remove_padding,
+            basecls=self.basecls
+        )
+
+        dnv = []
+        for dn, dv in six.iteritems(d):
+            if (dv is None or
+                    (isinstance(dv, (list, dict, set)) and len(dv) == 0)):
```

Comment:
you could use `hasattr(dv, "__iter__")` to achieve the same, but more general

Reply:
then, I need to add a try and except TypeError
```
>>> hasattr(ISOTP, "__iter__")                                                  
True
>>> hasattr(ISOTP, "__len__")                                                   
True
>>> len(ISOTP)                                                                  
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-9-e63da5f1917e> in <module>
----> 1 len(ISOTP)

TypeError: object of type 'Packet_metaclass' has no len()
```

Reply:
Sorry, I misunderstood your use case.. I thought you wanted to test for containers

Reply:
I do, that's why I like your solution. I will implement it with the stated try, except.

Reply:
or exclude if it is instance of `Packet`

Reply:
or derives from metaclass for Packet